packages/amplify-schema-validator/src/validators/relation-required-with-belongs-to.ts

```diff
@@ -24,10 +24,14 @@ export const validateRequireBelongsToRelation = (schema: DocumentNode): Error[]
       (directive) => directive.name.value === 'belongsTo',
     ));
     belongsToFields?.forEach((belongsToField) => {
-      const typeName = (belongsToField.type as NamedTypeNode)?.name.value;
+      const typeName = resolveFieldTypeName(belongsToField.type);
+      if (!typeName) {
+        /* istanbul ignore next */
+        return;
+      }
       const objectOfType = getObjectWithName(schema, typeName);
       const relationField = objectOfType?.fields?.find(
-        (field) => (field.type as NamedTypeNode)?.name?.value === objectName,
```

Comment:
the primary change in this PR is replacing this line, which assumes a `NamedTypeNode` to instead call a function to get the type name and works for `ListTypeNode` and `NonNullTypeNode` objects as well.